stable_baselines3/common/off_policy_algorithm.py

```diff
@@ -250,11 +239,17 @@ def learn(
         callback.on_training_start(locals(), globals())
 
         while self.num_timesteps < total_timesteps:
+            if isinstance(self.train_freq, int):
```

Comment:
missing comment

Reply:
but I would do that transform directly in the constructor, no?

Reply:
You mean I should add a comment describing the translation from the old style to the new style?

I deliberately kept the transition here since otherwise we would 1. need to store `n_steps` and `n_episodes` in the `OffPolicyAlgorithm` and 2. the user could not change the `train_freq` in hindsight or during training. Its unlikely that it is needed but WHEN you need it it is really annoying.

Reply:
>You mean I should add a comment describing the translation from the old style to the new style?

yes

>1. need to store n_steps and n_episodes in the OffPolicyAlgorithm 

this is already the case (`self.n_episodes_rollout` and `self.train_freq`)

> 2. the user could not change the train_freq in hindsight or during training.

it can if we store it.
Also, if needed, the user can direct call `collect_rollout` alone if needed.

Last thing, I think the check can be simplified if we have an assert about `self.train_freq[1]` before (see comment above ;))

Reply:
nope `self.n_episodes_rollout` is gone now. only `self.train_freq` is left and it is split up into old-style `n_steps` and `n_episodes` before passing it to `collect_rollouts` since I was too lazy to also change that implementation too (but we should do that in the future).

I did not quite get what you mean with the user calling `collect_rollout` directly.

Good point about normalizing `self.train_freq` to be a tuple in the constructor. I will do that.

Reply:
what I meant was keeping `self.n_episodes_rollout` (that would be nice to load old models too) but computing the value of `self.train_freq` and `self.n_episodes_rollout` in the constructor.

>I did not quite get what you mean with the user calling collect_rollout directly.

the same way the user can call `model.train()` (just updating the model), the user can also directly call `model.collect_rollout()` which the parameters the user wants.

Reply:
I would be reluctant to add the public `self.n_episodes_rollout` back. This way we basically build an API with two ways to specify the number of epochs between two training steps and introduce back the issue that the value of `self.n_episodes_rollout` can conflict with `self.train_freq`. If you like I can come up with a separate solution to ensure backwards compatibility with old stored models.

Since, as you pointed out, `model.collect_rollout()` is part of the public API and it still has the old-style `n_episodes` and `n_steps` interface that one should actually be updated to the new tuple format as well and I should stop being lazy!

Reply:
>I would be reluctant to add the public self.n_episodes_rollout back

Fair enough ;)

>nterface that one should actually be updated to the new tuple format as well and I should stop being lazy!

if you do so, you may also need to remove the `replay_buffer` parameter ;) (which was here for historic reasons)

Reply:
noted ...